src/app/app.component.ts

```diff
@@ -127,5 +174,34 @@ export class AppComponent {
         console.log(e);
         this.setStatus('Error sending coin; see log.');
       });
+
+    this.pong.deployed.sendCoin(receiver, amount, {from: this.pong.getYourAccount()});
+    this._ethObservable.refresh(this._pongService);
   };
 }
+export enum ContractEnum {
```

Comment:
Can we have these enums, interfaces and classes in their own separate files?

Reply:
It's one global ContractEnum. Every smart contract must have, one service, one interface ```*Deployed``` its the same what is called in your code ``` let meta = instance  ```, one``` *StaticData ``` and one class  implements ```ContractValues<*Deployed, *StaticData>``` So, yes we can have in separate files. I decided to store it, in one file because, I think it's more readable.